sim/main_game.py

```diff
@@ -15,13 +19,16 @@ def __init__(self):
         """
         pygame.init()
 
-        self.settings = Settings()
+        self.settings = GameSettings()
+        self.input_box_settings = InputBox()
 
         self.screen = pygame.display.set_mode(
             (self.settings.screen_width, self.settings.screen_height)
         )
         pygame.display.set_caption("ProjectG")
         self.player = Player(self)
+        self.type_box = InputBox(self)
```

Comment:
this class does not take an argument in the constructor, the Player + Npc do, but is self there correct? that is they take the instance of projectG as their first arg?